src/components/TrustedTimestamping/TimestampingPaginatedTable.vue

```diff
@@ -0,0 +1,170 @@
+<template lang="pug">
+div.row
+  div.col-12
+    card(card-body-classes="table-full-width" no-footer-line)
+      template(v-slot:header)
+        h4.card-title Timestamped Files
+
+      div
+        div.col-12.d-flex.justify-content-center.justify-content-sm-between.flex-wrap  
+          div.ml-auto
+            fg-input
+              el-input(
+                type="search"
+                class="mb-3"
+                clearable
+                prefix-icon="el-icon-search"
+                style="width: 200px"
+                placeholder="Search records"
+                v-model="searchQuery"
+                aria-controls="datatables"
+              )
+  
+        el-table(stripe style="width: 100%" :data="queriedData")
+          el-table-column(
+            v-for="column in tableColumns"
+            :key="column.label"
+            :min-width="column.minWidth"
+            :prop="column.prop"
+            :label="column.label"
+          )
+          
+          el-table-column(fixed="right" label="Actions")
+            template(v-slot:default="props")
+              div.d-flex.justify-content-center.table-actions
+                n-button(
+                  @click="handleCopy(props.$index, props.row)"
+                  type="info"
+                  size="sm"
+                  round
+                  icon
+                )
+                  i.fa.fa-copy
+        
+      template(v-slot:footer)
+        div.col-12.d-flex.align-items-center.justify-content-center
+          div
+            p.card-category Showing {{ from + 1 }} to {{ to }} of {{ total }} entries
+
+          div.ml-auto.d-flex.align-items-center
+            el-select.select-primary.mb-3.w-50(
+              v-model="pagination.perPage"
+              placeholder="Per page"
+            )
+              el-option.select-default(
+                v-for="item in pagination.perPageOptions"
+                :key="item"
+                :label="item"
+                :value="item"
+              )
+              
+            n-pagination.pagination-no-border(
+              v-model="pagination.currentPage"
+              :per-page="pagination.perPage"
+              :total="total"
+            )
+</template>
+<script>
+import { mapState } from "vuex";
+import { ElTable, ElTableColumn, ElSelect, ElOption } from "element-plus";
```

Comment:
@whatl3y Saw you imported some of these into the "reusable" components in the `faas` branch. Will likely add the components not imported yet into the "reusable" and be able to remove these imports here.